### DISCUSSION TITLE
How to skip bad elements when parsing multi-element JSON strings?

### DISCUSSION THREAD

**Discussion by usvi** [nlohmann/json#3141] [Q&A] [answered]
Hello,

Consider something like this:

'{"data": [ {"key": "data1", "value": "45", "type": "i"},{"key": "data2", "value": "", "type": "i"}  ]}'

I'm trying to parse that with json::parse . However, it throws error.

Basically parsing the following works, obviously:

'{"data": [ {"key": "data1", "value": "45", "type": "i"},{"key": "data2", "value": "3", "type": "i"}  ]}'

We already fixed the sender of bad data, but wouldn't do bad to be prepared. So, how could I make the parse "error resistant" on single bad elements? How could I keep only the elements where parse succeeds? There can be bad elements after and before good elements, randomly. The strings though are fixed. I see no point in trying to find some obscure regexp for splitting the data; my humble thought is that the library should manage it.

**Comment by gregmarr** [comment, marked as answer]
Can you provide sample code that reproduces the error?  I don't see anything there that would cause a parse error.

**Reply by usvi** [reply]
Yes, I will write specific case code today after my days meetings.

**Reply by usvi** [reply]
Example code showed that I had my own error when trying to stol() invalid string data. Sorry!

(To clarify my shame: I'm a full-blown C99 microcontroller backgrounded developer with my latest C++ experience from university some 10 years ago : P )
